azurelinuxagent/common/protocol/hostplugin.py

```diff
@@ -22,19 +22,24 @@
 
 HOST_PLUGIN_PORT = 32526
 URI_FORMAT_GET_API_VERSIONS = "http://{0}:{1}/versions"
+URI_FORMAT_GET_EXTENSION_ARTIFACT = "http://{0}:{1}/extensionArtifact"
 URI_FORMAT_PUT_VM_STATUS = "http://{0}:{1}/status"
 URI_FORMAT_PUT_LOG = "http://{0}:{1}/vmAgentLog"
 API_VERSION = "2015-09-01"
+HEADER_CONTAINER_ID = "x-ms-containerid"
+HEADER_VERSION = "x-ms-version"
+HEADER_HOST_CONFIG_NAME = "x-ms-host-config-name"
 
 
 class HostPluginProtocol(object):
-    def __init__(self, endpoint):
+    def __init__(self, endpoint, goal_state):
```

Comment:
I think I prefer that we pass the container_id and role_instance_config_name vs. the entire goal state. The more data passed, the more tightly the code is bound together. It may not matter much here, but it's a good practice to keep the binding between objects limited to what they need to know.